File: Ent_UsAdmin/Ent_UsAdmin/Agregar_Usuario.vb
Imports System.Data.OleDb
Imports System.Data.SqlClient
Imports System.Text

Public Class Agregar_Usuario
    Dim actualizar As Boolean = False
    Public autorizado As Boolean
    Public nombre, usuario, contraseña, idusuario, pass, passdecrypt As String
    Public grupo As String
    Public activo As Boolean
    Public fecha_alta As Date
    Dim cantidadgrupos As Integer
    Dim nombregrupo(1000) As String
    Dim codigogrupo(1000) As String
    Dim idgrupo(1000) As String
    Dim mostrarcontraseña As Boolean = False


    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = Windows.Forms.MouseButtons.Left Then
            MoveForm(Me)
        End If
    End Sub






    Private Sub Editar_Usuario_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        timerformato.Enabled = True
        cargargrupos()
        ConsultarEmpleado()

        dateusr.Value = Today

    End Sub

    Public Sub cargargrupos()
        Try
            Dim strgrupos As String
            strgrupos = "select id,nm,cdg from grp"
            Dim ejec = New OleDbCommand(strgrupos)
            ejec.Connection = conexionempresaR
            Dim labelgrupo As Label

            Dim numero As Integer = 0

            Dim myreadergrupos As OleDbDataReader = ejec.ExecuteReader()
            While myreadergrupos.Read
                labelgrupo = New Label

                idgrupo(numero) = myreadergrupos("id")
                codigogrupo(numero) = myreadergrupos("cdg")

                nombregrupo(numero) = myreadergrupos("nm")
                labelgrupo.Name = idgrupo(numero)
                labelgrupo.AutoSize = True
                labelgrupo.Text = idgrupo(numero) & "-" & codigogrupo(numero) & "-" & nombregrupo(numero)
                labelgrupo.ForeColor = Color.White
                AddHandler labelgrupo.Click, AddressOf clickevent
                FlowLayoutPanel1.Controls.Add(labelgrupo)
                numero += 1
                cantidadgrupos = numero
            End While
            myreadergrupos.Close()
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        FlowLayoutPanel1.Location = New Point(lblgrupo.Location.X, lblgrupo.Location.Y - lblgrupo.Height - FlowLayoutPanel1.Height)
        If FlowLayoutPanel1.Visible = False Then
            FlowLayoutPanel1.Visible = True
            FlowLayoutPanel1.BringToFront()
        Else
            FlowLayoutPanel1.Visible = False
        End If


    End Sub




    Private Sub timerformato_Tick(sender As Object, e As EventArgs) Handles timerformato.Tick
        timerformato.Interval = 1
        If mostrarcontraseña = False Then
            txtcontra.isPassword = True
        Else
            txtcontra.isPassword = False
        End If


    End Sub

    Private Sub txtempleado_OnValueChanged(sender As Object, e As EventArgs) Handles txtempleado.OnValueChanged

    End Sub

    Private Sub txtgrupo_TextChanged(sender As Object, e As EventArgs) Handles txtgrupo.TextChanged
        For numero As Integer = 0 To cantidadgrupos
            If txtgrupo.Text = idgrupo(numero) Then
                lblcdggrp.Text = codigogrupo(numero)
                lblnmgrp.Text = nombregrupo(numero)
                Exit For
            Else
                lblcdggrp.Text = "No existe"
                lblnmgrp.Text = "No existe"

            End If
        Next
    End Sub
    Public Sub clickevent(ByVal sender As Object, ByVal e As EventArgs)

        ' Dim label = DirectCast(sender, Label)

        txtgrupo.Text = sender.name
        FlowLayoutPanel1.Visible = False





    End Sub

    Private Sub Label5_Click(sender As Object, e As EventArgs) Handles Label5.Click
        If mostrarcontraseña = False Then
            Label5.BorderStyle = BorderStyle.Fixed3D
            Label5.BackColor = Color.FromArgb(223, 223, 223)
            mostrarcontraseña = True
        Else
            Label5.BorderStyle = BorderStyle.None
            Label5.BackColor = Color.DimGray
            mostrarcontraseña = False
        End If
    End Sub


    Private Sub btn_actualizar_Click(sender As Object, e As EventArgs) Handles btn_agregar.Click
        Autorizacion.tipoAutorizacion = "SatiModUsuariosAgregar"
        Autorizacion.ShowDialog()
        If autorizado Then
            Try
                Dim EncryptedMessage As Encryption.RSAResult = Encryption.RSA.Encrypt(txtcontra.Text, "<RSAKeyValue><Modulus>1D3WnN4qZq3DGHf7i3WjEixX0Nzka28RIR9lOrT7Eg2QmJTACm/E6388wMdGgH3yTwKPp/T5zm61yvZn44oHMka8EKlACGKkADsxN/ci0RLKJ0ZGYDVPtB+KSzI+GbmX2eO00R+FlYvXTpMzGXy3e4QpeCJbIbrDGFpt3rmXy28=</Modulus><Exponent>AQAB</Exponent></RSAKeyValue>")
                Dim contraseña As String
                Dim activosql As Integer
                contraseña = EncryptedMessage.AsBase64String
                Dim sql As String
                If Switchactivo.Value = False Then
                    activosql = 0
                Else
                    activosql = 1
                End If


                'sql = "update usr set nm = '" & txtusuario.Text & "',nm_complete='" & txtnombre.Text & "',pass='" & contraseña & "',grp='" & txtgrupo.Text & "',fecha_alta='" & dateusr.Value & "',activo='" & activosql & "' where idusr = '" & idusuario & "'"

                sql = "insert into usr(nm,pass,grp,nm_complete,addr,imgstr,tlf,fecha_alta,activo) values('" & txtusuario.Text & "','" & contraseña & "','" & txtgrupo.Text & "','" & txtnombre.Text & "','','','','" & dateusr.Value & "','" & activosql & "')"
                Dim comandoagregar As OleDbCommand

                iniciarconexionR()
                comandoagregar = New OleDbCommand
                comandoagregar.Connection = conexionempresaR
                comandoagregar.CommandText = sql
                comandoagregar.ExecuteNonQuery()

                If txtempleado.Text <> "0" Then
                    iniciarconexionempresa()
                    Dim comandoAempleado As SqlCommand
                    Dim consultaAempleado As String
                    consultaAempleado = "IF NOT EXISTS((select * from empleadosAsignados where usr = '" & txtusuario.Text & "' and idempleado = '" & txtempleado.Text & "'))
BEGIN
insert into empleadosAsignados values('" & txtusuario.Text & "','" & txtempleado.Text & "') end"
                    comandoAempleado = New SqlCommand
                    comandoAempleado.Connection = conexionempresa
                    comandoAempleado.CommandText = consultaAempleado
                    comandoAempleado.ExecuteNonQuery()

                Else

                End If


                actualizar = True



                MessageBox.Show("Listo")
            Catch ex As Exception
                actualizar = False
                MessageBox.Show(ex.Message)
            End Try
        Else
            MessageBox.Show("No fue autorizado")
        End If


    End Sub

    Public Sub ConsultarEmpleado()
        If txtempleado.Text <> "0" Then
            iniciarconexionempresa()
            Dim comandoEmpleado As SqlCommand
            Dim consultaEmpleado As String
            consultaEmpleado = "Select nombre from empleados where id = '" & txtempleado.Text & "'"
            comandoEmpleado = New SqlCommand
            comandoEmpleado.Connection = conexionempresa
            comandoEmpleado.CommandText = consultaEmpleado
            lblempleado.Text = comandoEmpleado.ExecuteScalar

        Else
            lblempleado.Text = "Ninguno"
        End If

    End Sub

    Private Sub txtempleado_KeyDown(sender As Object, e As KeyEventArgs) Handles txtempleado.KeyDown
        If e.KeyCode = Keys.Enter Then
            ConsultarEmpleado()

        End If

        If e.KeyCode = Keys.F2 Then
            BuscarEmpleadoUsuario.Show()
        End If
    End Sub
End Class

File: Ent_UsAdmin/Catalogos/Agregar_Usuario.cs
using System;
using System.Data.OleDb;
using System.Data.SqlClient;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace ConfiaAdmin
{

    public partial class Agregar_Usuario
    {
        private bool actualizar = false;
        public bool autorizado;
        public string nombre, usuario, contraseña, idusuario, pass, passdecrypt;
        public string grupo;
        public bool activo;
        public DateTime fecha_alta;
        private int cantidadgrupos;
        private string[] nombregrupo = new string[1001];
        private string[] codigogrupo = new string[1001];
        private string[] idgrupo = new string[1001];
        private bool mostrarcontraseña = false;

        public Agregar_Usuario()
        {
            InitializeComponent();
        }


        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                Module_resize.MoveForm(this);
            }
        }






        private void Editar_Usuario_Load(object sender, EventArgs e)
        {
            timerformato.Enabled = true;
            cargargrupos();
            ConsultarEmpleado();

            dateusr.Value = DateTime.Today;

        }

        public void cargargrupos()
        {
            try
            {
                string strgrupos;
                strgrupos = "select id,nm,cdg from grp";
                var ejec = new OleDbCommand(strgrupos);
                ejec.Connection = Module1.conexionempresaR;
                Label labelgrupo;

                int numero = 0;

                var myreadergrupos = ejec.ExecuteReader();
                while (myreadergrupos.Read())
                {
                    labelgrupo = new Label();

                    idgrupo[numero] = Conversions.ToString(myreadergrupos["id"]);
                    codigogrupo[numero] = Conversions.ToString(myreadergrupos["cdg"]);

                    nombregrupo[numero] = Conversions.ToString(myreadergrupos["nm"]);
                    labelgrupo.Name = idgrupo[numero];
                    labelgrupo.AutoSize = true;
                    labelgrupo.Text = idgrupo[numero] + "-" + codigogrupo[numero] + "-" + nombregrupo[numero];
                    labelgrupo.ForeColor = Color.White;
                    labelgrupo.Click += clickevent;
                    FlowLayoutPanel1.Controls.Add(labelgrupo);
                    numero += 1;
                    cantidadgrupos = numero;
                }
                myreadergrupos.Close();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            FlowLayoutPanel1.Location = new Point(lblgrupo.Location.X, lblgrupo.Location.Y - lblgrupo.Height - FlowLayoutPanel1.Height);
            if (FlowLayoutPanel1.Visible == false)
            {
                FlowLayoutPanel1.Visible = true;
                FlowLayoutPanel1.BringToFront();
            }
            else
            {
                FlowLayoutPanel1.Visible = false;
            }


        }




        private void timerformato_Tick(object sender, EventArgs e)
        {
            timerformato.Interval = 1;
            if (mostrarcontraseña == false)
            {
                txtcontra.isPassword = true;
            }
            else
            {
                txtcontra.isPassword = false;
            }


        }

        private void txtempleado_OnValueChanged(object sender, EventArgs e)
        {

        }

        private void txtgrupo_TextChanged(object sender, EventArgs e)
        {
            for (int numero = 0, loopTo = cantidadgrupos; numero <= loopTo; numero++)
            {
                if ((txtgrupo.Text ?? "") == (idgrupo[numero] ?? ""))
                {
                    lblcdggrp.Text = codigogrupo[numero];
                    lblnmgrp.Text = nombregrupo[numero];
                    break;
                }
                else
                {
                    lblcdggrp.Text = "No existe";
                    lblnmgrp.Text = "No existe";

                }
            }
        }
        public void clickevent(object sender, EventArgs e)
        {
            Label l = (Label)sender;
            // Dim label = DirectCast(sender, Label)

            txtgrupo.Text = Conversions.ToString(l.Name);
            FlowLayoutPanel1.Visible = false;





        }

        private void Label5_Click(object sender, EventArgs e)
        {
            if (mostrarcontraseña == false)
            {
                Label5.BorderStyle = BorderStyle.Fixed3D;
                Label5.BackColor = Color.FromArgb(223, 223, 223);
                mostrarcontraseña = true;
            }
            else
            {
                Label5.BorderStyle = BorderStyle.None;
                Label5.BackColor = Color.DimGray;
                mostrarcontraseña = false;
            }
        }


        private void btn_actualizar_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.Autorizacion.tipoAutorizacion = "SatiModUsuariosAgregar";
            My.MyProject.Forms.Autorizacion.ShowDialog();
            if (autorizado)
            {
                try
                {
                    var EncryptedMessage = Encryption.RSA.Encrypt(txtcontra.Text, "<RSAKeyValue><Modulus>1D3WnN4qZq3DGHf7i3WjEixX0Nzka28RIR9lOrT7Eg2QmJTACm/E6388wMdGgH3yTwKPp/T5zm61yvZn44oHMka8EKlACGKkADsxN/ci0RLKJ0ZGYDVPtB+KSzI+GbmX2eO00R+FlYvXTpMzGXy3e4QpeCJbIbrDGFpt3rmXy28=</Modulus><Exponent>AQAB</Exponent></RSAKeyValue>");
                    string contraseña;
                    int activosql;
                    contraseña = EncryptedMessage.AsBase64String;
                    string sql;
                    if (Switchactivo.Value == false)
                    {
                        activosql = 0;
                    }
                    else
                    {
                        activosql = 1;
                    }


                    // sql = "update usr set nm = '" & txtusuario.Text & "',nm_complete='" & txtnombre.Text & "',pass='" & contraseña & "',grp='" & txtgrupo.Text & "',fecha_alta='" & dateusr.Value & "',activo='" & activosql & "' where idusr = '" & idusuario & "'"

                    sql = "insert into usr(nm,pass,grp,nm_complete,addr,imgstr,tlf,fecha_alta,activo) values('" + txtusuario.Text + "','" + contraseña + "','" + txtgrupo.Text + "','" + txtnombre.Text + "','','','','" + Conversions.ToString(dateusr.Value) + "','" + activosql + "')";
                    OleDbCommand comandoagregar;

                    Module1.iniciarconexionR();
                    comandoagregar = new OleDbCommand();
                    comandoagregar.Connection = Module1.conexionempresaR;
                    comandoagregar.CommandText = sql;
                    comandoagregar.ExecuteNonQuery();

                    if (txtempleado.Text != "0")
                    {
                        Module1.iniciarconexionempresa();
                        SqlCommand comandoAempleado;
                        string consultaAempleado;
                        consultaAempleado = "IF NOT EXISTS((select * from empleadosAsignados where usr = '" + txtusuario.Text + "' and idempleado = '" + txtempleado.Text + @"'))
BEGIN
insert into empleadosAsignados values('" + txtusuario.Text + "','" + txtempleado.Text + "') end";
                        comandoAempleado = new SqlCommand();
                        comandoAempleado.Connection = Module1.conexionempresa;
                        comandoAempleado.CommandText = consultaAempleado;
                        comandoAempleado.ExecuteNonQuery();
                    }

                    else
                    {

                    }


                    actualizar = true;



                    MessageBox.Show("Listo");
                }
                catch (Exception ex)
                {
                    actualizar = false;
                    MessageBox.Show(ex.Message);
                }
            }
            else
            {
                MessageBox.Show("No fue autorizado");
            }


        }

        public void ConsultarEmpleado()
        {
            if (txtempleado.Text != "0")
            {
                Module1.iniciarconexionempresa();
                SqlCommand comandoEmpleado;
                string consultaEmpleado;
                consultaEmpleado = "Select nombre from empleados where id = '" + txtempleado.Text + "'";
                comandoEmpleado = new SqlCommand();
                comandoEmpleado.Connection = Module1.conexionempresa;
                comandoEmpleado.CommandText = consultaEmpleado;
                lblempleado.Text = Conversions.ToString(comandoEmpleado.ExecuteScalar());
            }

            else
            {
                lblempleado.Text = "Ninguno";
            }

        }

        private void txtempleado_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                ConsultarEmpleado();

            }

            if (e.KeyCode == Keys.F2)
            {
                My.MyProject.Forms.BuscarEmpleadoUsuario.Show();
            }
        }
    }
}